### PR TITLE
(Re)Adds a new shotgun scout specialization role.

### PR DESCRIPTION
**Pull request by BurgerLUA**
image

## About The Pull Request
Adds a new shotgun specialization kit. It contains what you'd typically find in the scout set, except a different weapon, thermal imaging goggles, and a M1911 service pistol.

The different weapon is a unique .410 caliber lever action shotgun. It stores 5 rounds, and shoots pretty fast if you pump it quickly. So far it comes with a sabot ammo type, a buckshot ammo type, and a special "tracker" ammo type that leaks blue droplets every 4 seconds for 80 seconds on the target you shoot at. Multiple shots landed increase the duration. 

## Why It's Good For The Game
I think a shotgun specialization was lacking, and there is a problem with xenos hiding nearly every round after performing hit and run attacks. This kit should counter that sort of behavior.

## Changelog
:cl: BurgerBB
add: Adds a new shotgun specialization kit for specialists.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/guns/shotguns.dm [hunk 1]
```diff
@@ -653,3 +659,51 @@ can cause issues with ammo types getting mixed up during the burst.
 
 /obj/item/weapon/gun/shotgun/pump/lever/pump_notice(mob/user)
 	to_chat(user, "<span class='notice'><b>You work [src] lever.</b></span>")
+
+
+/obj/item/weapon/gun/shotgun/pump/lever/mbx900
+	name = "\improper MBX-900 Lever Action Rifle"
```
Comment (Hughgent) [inline code/modules/projectiles/updated_projectiles/guns/shotguns.dm]: 'Rifled Shotgun' might be a more apt description?

**Comment by spookydonut** [comment]
hahahahahahahaha

**Comment by BurgerLUA** [comment]
y u laugh :(

**Comment by TiviPlus** [comment]
Ahhh shit here we go again (That stun better stay gone)

**Comment by BurgerLUA** [comment]
I don't think any of the shotgun rounds that I added have a stun function.

**Comment by BurgerLUA** [comment]
As I said on the developer discord, I cannot do the requested RegisterSignal method of tracking unless you want me to implement some sort of weird framework that involves placing an actual object inside a mob that runs its own timer event. There is very little documentation on RegisterSignal, and given the documentation that exists, it feels inappropriate to use it in this instance.

File: code/modules/projectiles/updated_projectiles/magazines/shotguns.dm [hunk 2]
```diff
@@ -51,6 +51,32 @@ one type of shotgun ammo, but I think it helps in referencing it. ~N
...
+
+/obj/item/ammo_magazine/mbx900/tracking
+	name = ".410 Tracker Ammo Box"
+	desc = "A box filled with .410 tracker rounds."
+	icon_state = "mbx900_tracker"
+	default_ammo = /datum/ammo/bullet/shotgun/mbx900_tracker
+
+
+
+/datum/ammo/bullet/shotgun/mbx900_tracker
```
Comment (factoryman942) [inline code/modules/projectiles/updated_projectiles/magazines/shotguns.dm]: ??

File: code/modules/projectiles/updated_projectiles/magazines/shotguns.dm [hunk 3]
```diff
@@ -51,6 +51,32 @@ one type of shotgun ammo, but I think it helps in referencing it. ~N
 	max_rounds = 20 // Real rifle boxes are usually 20 rounds. This works with the new system, four handfuls.
 	w_class = WEIGHT_CLASS_SMALL // CAN throw it in your pocket, friend.
 
+
+/obj/item/ammo_magazine/mbx900/
```
Comment (factoryman942) [inline code/modules/projectiles/updated_projectiles/magazines/shotguns.dm]: should these maybe be:
```suggestion
/obj/item/ammo_magazine/shotgun/mbx900
```

**Comment by factoryman942** [comment]
- Why an M1911 and not an 88mod4 like other specs?
- Do you expect the tracker shot / motion tracker to get much use when the kit has thermals, instead of just normal nightvis?
- Why are you reusing ammo_hud icon states? reeeee

**Comment by BurgerLUA** [comment]
> * Why an M1911 and not an 88mod4 like other specs?
> * Do you expect the tracker shot / motion tracker to get much use when the kit has thermals, instead of just normal nightvis?
> * Why are you reusing ammo_hud icon states? reeeee

- M1911 is an american classic. So is a lever action shotgun.
- Motion tracker has its pros and cons. You don't have to use the motion tracker or the nightvis sight if - 
 you don't want to, and switch it for something else.
- There is nothing wrong with reusing icon HUD icon states.

File: code/game/objects/effects/decals/Cleanable/humans.dm [hunk 4]
```diff
@@ -106,6 +106,18 @@
...
 	var/drips
 
+
+/obj/effect/decal/cleanable/blood/drip/tracking_fluid
+	name = "tracking fluid"
+	desc = "Tracking fluid from a tracking round."
+	basecolor = "#00FFFF"
+
+/obj/effect/decal/cleanable/blood/drip/tracking_fluid/dry()
+	name = "dried [src.name]"
```
Comment (Rohesie) [inline code/game/objects/effects/decals/Cleanable/humans.dm]:
```suggestion
	name = "dried [name]"
```

**Comment by factoryman942** [comment]
> * There is nothing wrong with reusing icon HUD icon states.

well one of them's the wrong colour for one thing <s> literally just bully kuro for sprites</s>
also, reusing stuff means that if/when it gets changed it'll possibly become "wrong" for this weapon; particularly using AK and mosin sounds for a shotgun

**Comment by BurgerLUA** [comment]
It's in line with the code that you guys already have. A lot of your weapons reuse the same cocking sounds and shooting sounds. I'll add a new HUD sprite for the black shells.

**Comment by TheBimmer231** [comment]
the obj sprite needs some work, ping me on discord.

File: code/datums/supplypacks.dm [hunk 5]
```diff
@@ -1792,6 +1802,20 @@ AMMO
 	containername = "\improper incendiary sniper ammo crate"
 	group = "Ammo"
 
+/datum/supply_packs/ammo_mbx900
+	name = "MBX 900 ammo crate (x2 sabot, x2 buckshot, x1 tracking)"
+	contains = list(
+					/obj/item/ammo_magazine/shotgun/mbx900/,
```
Comment (Rohesie) [inline code/datums/supplypacks.dm]:
```suggestion
					/obj/item/ammo_magazine/shotgun/mbx900,
```

File: code/datums/supplypacks.dm [hunk 6]
```diff
@@ -1792,6 +1802,20 @@ AMMO
...
+					/obj/item/ammo_magazine/shotgun/mbx900/,
```
Comment (Rohesie) [inline code/datums/supplypacks.dm]:
```suggestion
					/obj/item/ammo_magazine/shotgun/mbx900,
```

File: code/modules/clothing/glasses/thermal.dm [hunk 7]
```diff
@@ -54,3 +54,11 @@
 	icon_state = "thermalimplants"
 	item_state = "syringe_kit"
 	toggleable = 0
+
+/obj/item/clothing/glasses/thermal/m64_thermal_goggles
```
Comment (Rohesie) [inline code/modules/clothing/glasses/thermal.dm]: This is pretty powerful and a balance concern.

File: code/modules/mob/living/living.dm [hunk 8]
```diff
@@ -24,6 +24,13 @@
 	handle_drugged()
 	handle_stuttering()
 	handle_slurring()
+	handle_tracking()
+
+/mob/living/proc/handle_tracking()
```
Comment (Rohesie) [inline code/modules/mob/living/living.dm]: This can go now.

File: code/modules/mob/living/living_defines.dm [hunk 9]
```diff
@@ -24,6 +24,7 @@
 	var/is_dizzy = FALSE
 	var/druggy = 0
 	var/sleeping = 0
+	var/track_amount = 0
```
Comment (Rohesie) [inline code/modules/mob/living/living_defines.dm]:
```suggestion
```

File: code/modules/projectiles/updated_projectiles/ammunition.dm [hunk 10]
```diff
@@ -210,7 +210,13 @@ If it is the same and the other stack isn't full, transfer an amount (default 1)
...
 
 	name = "handful of [ammo_name + (ammo_name == "shotgun buckshot"? " ":"s ") + "([new_caliber])"]"
-	icon_state = new_caliber == "12g" ? ammo_name : new_caliber == "7.62x54mmR" ? "mosin bullet" : "bullet" //What am I even doing with my life?
+	icon_state = "bullet"
+	switch(new_caliber)
+		if("12g",".410")
+			icon_state = ammo_name
+		if("7.62x54mmR")
+			icon_state = "mosin bullet"
+
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/ammunition.dm]:
```
else
	icon_state = "bullet"
```

File: code/modules/projectiles/updated_projectiles/magazines/shotguns.dm [hunk 11]
```diff
@@ -51,6 +51,29 @@ one type of shotgun ammo, but I think it helps in referencing it. ~N
 	max_rounds = 20 // Real rifle boxes are usually 20 rounds. This works with the new system, four handfuls.
 	w_class = WEIGHT_CLASS_SMALL // CAN throw it in your pocket, friend.
 
+
+/obj/item/ammo_magazine/shotgun/mbx900/
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/magazines/shotguns.dm]:
```suggestion
/obj/item/ammo_magazine/shotgun/mbx900
```

File: code/datums/components/dripping.dm [hunk 12]
```diff
@@ -0,0 +1,58 @@
+#define DRIP_ON_WALK 1
+#define DRIP_ON_TIME 2
+
+/datum/component/dripping
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	var/drips_left = 0
```
Comment (Rohesie) [inline code/datums/components/dripping.dm]:
```suggestion
```
Who'd be dumb enough to add a variable that does nothing?

File: code/datums/components/dripping.dm [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+	var/drip_time = 60 SECONDS
```
Comment (Rohesie) [inline code/datums/components/dripping.dm]:
```suggestion
```

File: code/modules/clothing/glasses/thermal.dm (hunk 7 above)
Comment (BurgerLUA) [reply code/modules/clothing/glasses/thermal.dm]: You won't get night vision from this, and most maps have a lot of glass windows or are wide open anyways so thermal vision would only work in some areas.

**Comment by BurgerLUA** [comment]
Done.

**Comment by Rohesie** [comment]
Use the new sprites Kuro made and we can test this.

**Comment by InterroLouis** [comment]
There's one change I'd like to see before this gets fully merged, if it does-and that's making the gun use the specialist weapon skill rather than the shotgun skill.

**Comment by BurgerLUA** [comment]
Closing this PR until the game is in a better state.

**Comment by Rohesie** [comment]
Check the icondiff. Some conflicts were badly solved.

**Comment by Rohesie** [comment]
Empty `.410 Sabot Ammo Box` sprite issue.

**Comment by Rohesie** [comment]
Is the `pulse_0` icon change intended?

**Comment by BurgerLUA** [comment]
> Is the `pulse_0` icon change intended?

yes because it will fuck up the dmi merge tool otherwise because there are 2 blank icon states and dmimerge doesn't know how to handle that

**Comment by BurgerLUA** [comment]
Increased the max capacity of the shotgun from 5 to 10.

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-3). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**